scyllacloud/hostdialer_test.go

```diff
@@ -0,0 +1,209 @@
+package scyllacloud
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+const (
+	testTimeout = time.Second
+)
+
+func TestHostSNIDialer_ServerNameIdentifiers(t *testing.T) {
+	t.Parallel()
+
+	connectionStateCh := make(chan tls.ConnectionState, 1)
+
+	server := httptest.NewUnstartedServer(nil)
+
+	clientCert, clientKey, err := generateClientCert([]string{"host-1-uuid.node.scylladb.com", "node.scylladb.com"})
```

Comment:
is this mixing client certs and serving certs?

Reply:
yes it was done to save time generating keys, but once i changed it to 1024 rsa it's fast enough to have two separate.

Reply:
talking speed, have you considered using a pregenerated cert, like in https://github.com/golang/go/blob/master/src/net/http/internal/testcert/testcert.go#L14-L63 (but embedded) ?

Reply:
yes, it gives speed, but it's limits flexibility. On my machine entire suite takes ~60ms without `-race` and ~200ms with `-race` so i went with generation on each run.